optaplanner-core/src/main/java/org/optaplanner/core/impl/partitionedsearch/DefaultPartitionedSearchPhaseFactory.java

```diff
@@ -76,13 +77,13 @@ public DefaultPartitionedSearchPhaseFactory(
         return phase;
     }
 
-    private SolutionPartitioner buildSolutionPartitioner() {
+    private SolutionPartitioner<Solution_> buildSolutionPartitioner() {
         if (phaseConfig.getSolutionPartitionerClass() != null) {
-            SolutionPartitioner<?> solutionPartitioner = ConfigUtils.newInstance(this,
-                    "solutionPartitionerClass", phaseConfig.getSolutionPartitionerClass());
+            SolutionPartitioner<?> solutionPartitioner =
+                    ConfigUtils.newInstance(phaseConfig, "solutionPartitionerClass", phaseConfig.getSolutionPartitionerClass());
```

Comment:
Why have you replaced `this` with `phaseConfig`?

Reply:
If you take a look at the `ConfigUtils.newInstance()`, the reference is used to properly communicate an error during instantiation of the `SolutionPartitionerClass`, which is a property of the `phaseConfig` object.